package/MDAnalysis/coordinates/base.py

```diff
@@ -1741,7 +1753,71 @@ def get_aux_descriptions(self, auxnames=None):
             auxnames = self.aux_list
         descriptions = [self._auxs[aux].get_description() for aux in auxnames]
         return descriptions
-
+    
+    def add_transformation(self, transname, transform):
+        """ Add a transformation to be applied to the trajectory.
+        
+        The transformations are part of the `~MDAnalysis.transformations` module
+        and are stored as a OrderedDict `transformations`. The transformations are
+        part of the `~MDAnalysis.transformations` module. Each transformation 
+        may or may not require arguments. Multiple transformations can be queued.
+        e.g. to translate the coordinate of the trajectory by the (1, 2, 3)
+        vector::
+        
+            u = MDAnalysis.Universe(PDB, XTC)
+            u.trajectory.add_transformation('translate', MDAnalysis.transformations.translate(1, 2, 3))
+        
+        This adds the 'translate' transformation and applies it to the current 
+        frame.
+        
+        Parameters
+        ----------
+        transname : str
+            Name of the transformation
+        transform
+            transformation to be applied
+        
+        See Also
+        --------
+        :meth:`remove_transformation`
+        """
+        self.transformations[transname] = transform
+        self.ts = self._check_and_apply_transformation(transname, self.ts)
+        # call reader here to apply the newly added transformation on the
+        # current loaded frame?
+    
+    def remove_transformation(self, transname):
+        """ Remove a transformation from the OrderedDict `transformations`
+        
+        Parameters
+        ----------
+        transname : str
+            Name of the transformation
+        
+        See Also
+        --------
+        :meth:`add_trasnformation`
+        """
+        try:
+            del self.transformations[transname]
+            self.ts.applied_transformations[self.frame].remove(transname)
+        except:
+            raise ValueError("No transformation named {name}".format(name=transname))
+        
+    def _check_and_apply_transformation(self, transname, ts):
+        """ Check if a given transformation name has already been applied to a
+        given timestep """
+        if ts.frame not in ts.applied_transformations.keys():
+            ts.applied_transformations[ts.frame] = [transname]
```

Comment:
Doesn't this assume that the timestep is the same instance from one frame to the other? I think we can assume that some readers may not recycle the TimeStep instance and use a new one each time a frame is read.

Reply:
You are right. When you read from a file the same frame over and over again, it will always be a new instance of ```Timestep``` - coordinates are reset, so the transformations are always applied. But for the memoryreader, as the coordinates are changed in the "source", this will avoid that problem.